### PR TITLE
Micro Reddit

### PR DESCRIPTION
**Pull request by GraceOyiza**
In this project we built micro-reddit using:
-Ruby
-Rails
-Active record
-Rubocop for checking linters.

### CODE REVIEW

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team 👋
Status: Changes Required ♻️

Congratulations team! 👏 Good job so far 👍😊
Let's cover a few details to make this project better, together ⏬

## [OPTIONAL] Ruby version
- .ruby-version file(ruby-2.7.0), Gemfile(ruby '2.7.0') and your Readme file(Ruby 2.7.1), they are notating different versions of the Ruby. They should be the same.

## [OPTIONAL] Database
- You have configured the database to run pg. You don't need to do that for this project. If you want to do it, please provide sufficient setup instructions in the readme file. Because the project is not working unless I change it to the sqlite3.

Screenshot 📸
image

## [REQUIRED] Readme
- [ ] The project has a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
💡 Please rewrite your Readme file so anyone can follow readme and make the application work.

## [REQUIRED] Commits
- [ ] Commits are descriptive ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_commits.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
💡 Please, commit more often.

## [REQUIRED] Models
💡 I can see that you are missing most of the requirements in models. This is totally fine, I am sure you will learn a lot from this project. Please be sure you read the documentation on our path.
Models are where we are configuring our entities. So it is important. This place is also where we add validations.

> Be sure you read the Odin project and follow the entries one by one. If you do so, you will do great at this project, and not only that, you will learn a lot. Here's a screenshot from the Odin project in which taking us on a journey and teaching step by step.

Screenshot 📸
image

> When I was learning the Rails with my coding partner, we were working with Pomodoro technique. We were switching roles every 45 minutes and the navigator was reading these entries from the Odin project. The driver was writing the necessary code.

> This part of Ruby on Rails is very important. Try to get comfortable with `rails console`. 😊🎉

> So, it is a good practice to follow the Odin project, line by line. From the beginning.

### [REQUIRED]  User
- [ ] username is validated (at least for the presence) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/userValidation.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [ ] `has_many :posts`([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/user_has_many_posts.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [ ] `has_many :comments`([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/user_has_many_comments.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))

### [REQUIRED] Post
- [ ] validations are present ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/postValidation.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [ ] `belongs_to :user`([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/comment_belongs_to_a_user.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))

### [REQUIRED] Comment
- [ ] validations are present ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/commentValidation.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [ ] `belongs_to :user` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/comment_belongs_to_a_user.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))

### That's it, team! After making the changes you can request another code review 😊

I am really wondering about the next version of your project ⭐😇
With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts
+  validates :name, presence: true
```
Comment (moinkhanif) [inline app/models/user.rb]: - [ ] **[REQUIRED]** The user model `has_many :comments` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/user_has_many_comments.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  validates :author, presence: true
```
Comment (moinkhanif) [inline app/models/comment.rb]: - [ ] **[REQUIRED]** The comment model `belongs_to :user`  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/comment_belongs_to_a_user.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))

- [ ] **[REQUIRED]** Presence of `post` is being checked two times. Usually `belongs_to` would be enough in this situation. When I try to save an invalid comment using rails console, I get this:
```ruby
irb(main):023:0> c1 = Comment.new
irb(main):024:0> c1.save
=> false
irb(main):025:0> c1.errors.full_messages
=> ["Post must exist", "Post can't be blank", "Author can't be blank", "Body can't be blank"]
```
Notice, that it's checking twice for post. Please fix it.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  has_many :comments
+  belongs_to :user
+  validates :title, presence: true
+  validates :body, presence: true
+  validates :user, presence: true
```
Comment (moinkhanif) [inline app/models/post.rb]: - [ ] **[REQUIRED]** Presence is being checked two times. Usually `belongs_to` would be enough in this situation. When I try to save an invalid post using rails console, I get this:
```ruby
irb(main):016:0> p1 = Post.new
irb(main):017:0> p1.save
=> false
irb(main):018:0> p1.errors.full_messages
=> ["User must exist", "User can't be blank", "Title can't be blank", "Body can't be blank"]
```
Notice, that its checking twice for user. Please fix it.

**Review by moinkhanif — CHANGES_REQUESTED** [review]
## Changes Required ❗

Hey team, 

Great job implementing some of the requested changes so far. You are almost done. Please fix the following changes requested in the comments below to fix all the issues related to your project:

Once, you have completed the requested changes, please request a review again.

Happy Coding!

**Comment by Dandush03** [comment]
# Status: Project Approved :rocket: 
### Great Work, Keep It Up!
# Happy Coding :computer:

**Pull request merged**